numpy/ma/core.py

```diff
@@ -6397,6 +6397,9 @@ def __repr__(self):
             # it's a subclass, or something is wrong, make it obvious
             return object.__repr__(self)
 
+    def __format__(self, format_spec):
+        return format(str(masked_print_option._display), format_spec)
```

Comment:
```suggestion
        # Replace ndarray.__format__ with the default, which supports no format characters.
        # Supporting format characters is unwise here, because we do not know what type
        # the user was expecting - better to not guess.
        return object.__format__(self, format_spec)
```